Data Preparation.R

```diff
@@ -0,0 +1,173 @@
+##################################################################
+## Name of File: End of Life All Activity                       ##
+## Original Authors : Federico Centoni/Alice Byers              ##
+## Originally written: November 2018                            ##                      
+## Latest Update Date: 23/07/2019.                              ##
+## Written/run on : R Studio Server                             ##
+## R version:                                                   ##
+## Package version:                                             ##
+## Description: Outputs All Activity file to the patient level  ##
+##              for use in EOL publication                      ##
+##################################################################
+
+
+
+###################################
+## SECTION 1 - HOUSEKEEPING      ##
+###################################
+
+install.packages("readxl")
+install.packages("here")
+install.packages("readr")
+install.packages("dplyr")
+install.packages("tidyr")
+install.packages("odbc")
+install.packages("janitor")
+
+## 1. Load packages ##
+
+library(odbc)      # for accessing SMRA
+library(dplyr)     # functions for 'tidy' data wrangling
+library(haven)     # read and write spss files
+library(lubridate) # functions for use with dates
+library(magrittr)  # for %<>% operator
+library(tidyverse)
+library(janitor)   
+library(tictoc)
+
+
+## 2. Define dates ##
+
+extract_date <- "_extracted_22072019"
+
+start_date        <- "2009-10-01"  # start date for SMR01 and SMR50 extracts
+end_date          <- "2018-03-31"  # end date for SMR01 and SMR50 extracts
+deaths_start_date <- "2010-04-01"  # start date for deaths extract
+
+## 3. Set filepaths ##
+
+# Output filepath
+output <- "/PHI_conf/irf/EOL_Publication/Data preparation/RAP/Data Preparation/"
+
+# Lookup filepaths
+pc_lookup   <- "/conf/linkage/output/lookups/Unicode/Geography/
+                Scottish Postcode Directory/
+                Scottish_Postcode_Directory_2019_1.5.sav"
+simd_lookup <- "/conf/linkage/output/lookups/Unicode/Deprivation/
+                postcode_2019_1.5_simd2016.sav"
+locality    <- "/isdsf00d03/cl-out/lookups/Unicode/Geography/HSCP Locality/
+                HSCP Localities_DZ11_Lookup_20180903.sav"
```

Comment:
You can use .rds files in the folder instead of using .sav files. It will make things easier so that you don't need to deal with SPSS files!